ResearchKit/Onboarding/ORKRegistrationStep.m

```diff
@@ -0,0 +1,258 @@
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGivenName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemGivenName
+                                                               text:ORKLocalizedString(@"CONSENT_NAME_GIVEN", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"GIVEN_NAME_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeFamilyName) {
+        ORKTextAnswerFormat *answerFormat = [ORKAnswerFormat textAnswerFormat];
+        answerFormat.multipleLines = NO;
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemFamilyName
+                                                               text:ORKLocalizedString(@"CONSENT_NAME_FAMILY", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"FAMILY_NAME_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+        
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeGender) {
+        NSArray *textChoices = @[[ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_MALE", nil) value:@0],
+                                 [ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_FEMALE", nil) value:@1],
+                                 [ORKTextChoice choiceWithText:ORKLocalizedString(@"GENDER_OTHER", nil) value:@2]];
+        ORKValuePickerAnswerFormat *answerFormat = [ORKAnswerFormat valuePickerAnswerFormatWithTextChoices:textChoices];
+        
+        ORKFormItem *item = [[ORKFormItem alloc] initWithIdentifier:ORKRegistrationFormItemGender
+                                                               text:ORKLocalizedString(@"GENDER_FORM_ITEM_TITLE", nil)
+                                                       answerFormat:answerFormat];
+        item.placeholder = ORKLocalizedString(@"GENDER_FORM_ITEM_PLACEHOLDER", nil);
+        item.optional = NO;
+       
+        [formItems addObject:item];
+    }
+    
+    if (_options & ORKRegistrationStepIncludeDOB) {
+        // Calculate default date (20 years from now).
+        NSDateComponents *minusTwentyYears = [NSDateComponents new];
+        minusTwentyYears.year = -20;
+        NSDate *defaultDate = [[NSCalendar currentCalendar] dateByAddingComponents:minusTwentyYears
+                                                                                toDate:[NSDate date]
+                                                                               options:0];
+        
+        ORKDateAnswerFormat *answerFormat = [ORKAnswerFormat dateAnswerFormatWithDefaultDate:defaultDate
```

Comment:
Not sure if the permissions page has been presented already at this point. I'm not certain if I want to grab anything from HK during the on boarding process. What do you think?